python/tk_multi_workfiles/actions/file_action_factory.py

```diff
@@ -36,158 +37,336 @@
 from ..file_item import FileItem
 from ..user_cache import g_user_cache
 
+
 class FileActionFactory(object):
     """
+    Builder for the file open menu. Based on the current selection, the build will enumerate a list of actions,
+    grouped actions and seperators that should be displayed in the UI.
     """
-    
-    def get_actions(self, file, work_area, file_model, workfiles_visible=True, publishes_visible=True):
+
+    def __init__(self, work_area, file_model, workfiles_visible=True, publishes_visible=True):
```

Comment:
The factor was being instantiated every single time we needed it, so the performance should be the same. Moving some of those variables to data members made the code cleaner since it's static state that doesn't need to be passed around to other methods anymore.